gateway-service-knoxssout/src/main/java/org/apache/knox/gateway/service/knoxsso/WebSSOutResource.java

```diff
@@ -106,6 +111,21 @@ private boolean removeAuthenticationToken(HttpServletResponse response) {
     }
     response.addCookie(c);
 
+    Cookie[] cookies = request.getCookies();
+    if (cookies != null) {
+      Optional<Cookie> SSOCookie = Arrays.stream(cookies).filter(cookie -> cookie.getName().equals(cookieName)).findFirst();
+      if (SSOCookie.isPresent()) {
+        GatewayServices gwServices =
+                (GatewayServices) request.getServletContext().getAttribute(GatewayServices.GATEWAY_SERVICES_ATTRIBUTE);
+        if (gwServices != null) {
+          ConcurrentSessionVerifier verifier = gwServices.getService(ServiceType.CONCURRENT_SESSION_VERIFIER);
+          verifier.sessionEndedForUser(request.getUserPrincipal().getName(), SSOCookie.get().getValue());
```

Comment:
Do we always have a ConcurrentSessionVerifier? I see at other places we check if it's null or not. Probably we need to add a null check here too.